src/file/format/hjson.rs

```diff
@@ -2,15 +2,25 @@ use serde_hjson;
 use source::Source;
 use std::collections::HashMap;
 use std::error::Error;
+use std::io;
 use value::{Value, ValueKind};
 
 pub fn parse(
     uri: Option<&String>,
     text: &str,
 ) -> Result<HashMap<String, Value>, Box<Error + Send + Sync>> {
     // Parse a JSON object value from the text
-    // TODO: Have a proper error fire if the root of a file is ever not a Table
-    let value = from_hjson_value(uri, &serde_hjson::from_str(text)?);
+    let root = serde_hjson::from_str(text)?;
+    match root {
+        serde_hjson::Value::Object(_) => (),
+        _ => {
+            return Err(Box::new(io::Error::new(
+                io::ErrorKind::InvalidData,
+                format!("Root of resource({:?}) is not a Table", uri),
+            )))
+        }
```

Comment:
You can remove this scope I guess.

Reply:
I just followed the original TODO, if it's not necessary, please let me know.